Add check before deleting directory

stfnx opened the pull request:
If the target directory is a mountpoint (eg. when using Impermanence), the command will fail because "Device or resource busy". So check if its a mountpoint beforehand.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

stfnx committed:
Add check before deleting directory

AndersonTorres commented:
"Add check before deleting directory" -  what? Which directory?

Please read the commit coding conventions:

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

AndersonTorres commented:
Also, avoid the repeated closing of pull requests.
(And this is why we do not like the usage of github web ui...)

stfnx commented:
I did not repeatedly close pull requests.
You closed the first one and I closed this one. Both for a reason.

I added a very simple change to the code. Literally one line.
Everyone will understand what this change does by looking at it for a few seconds.
The commit message explains the change well enough in my opinion.
And I added a (admittedly very small) description to this PR.

Apparently this is not enough. In which case I'm not interested in going any further.
That's why I closed this PR.

And I don't know what you issue is with the GitHub UI and why you need to mention it in any sentence. You chose GitHub as a platform so you should expect people to use it. This is one of the first pull requests of my life and I have not too much experience with Git/GitHub. I just used the tool that currently is the easiest for me.
